src/codegen/mod.rs

```diff
@@ -159,7 +160,14 @@ pub fn codegen(ns: &mut Namespace, opt: &Options) {
     if ns.target == Target::Solana {
         for contract_no in 0..ns.contracts.len() {
             if ns.contracts[contract_no].instantiable {
-                collect_accounts_from_contract(contract_no, ns);
+                let diag = collect_accounts_from_contract(contract_no, ns);
+                ns.diagnostics.extend(diag);
+            }
+        }
+
+        for contract_no in 0..ns.contracts.len() {
```

Comment:
Nitpick using iterators:
```rust
for (n,_) in ns.contracts.iter().enumerate().filter(|(_, c)| c.instantiable) {}
```

Reply:
This is not possible, because the iterator borrows `ns`, while the function call in line 170 borrows `ns` as mutable.

Reply:
So `iter_mut` should work?

Reply:
It doesn't work. I cannot borrow `ns` both in the for loop header and in its body.